### PR TITLE
feat: compression option with xz support for deb

### LINKED ISSUE #361: XZ-compressed Debian packages
I was reading over the Debian repository format, which lead to me looking into their current package formats and discovered that they now seem to require XZ support. Inspecting random packages in official Debian repos, I found the .deb  and the data.tar inside are both XZ-compressed, so it looks like they want this to be the default now.

There's already some support for creating .tar.xz archives in GoReleaser, so it's not a huge leap to also apply it to Debian packages. Making it default is probably not going to be problematic, but having the option to support old clients which only read Gzip is of course useful.

### PR DESCRIPTION
**Pull request by erikgeiser**
This PR adds a `compression` option for `deb` files. Currently the following values are supported:

* `gzip` (default)
* `xz` (future default)
* `none` (no compression)

I think the best way to move forward is to keep `gzip` as default for now and wait for feedback from the users that use `xz`. If everything works great (and if `xz` build times are acceptable), we should make `xz` the default in a following release.

Closes #361.

### CODE REVIEW

File: deb/deb.go [hunk 1]
```diff
@@ -90,7 +91,7 @@ func (*Deb) Package(info *nfpm.Info, deb io.Writer) (err error) { // nolint: fun
 		return err
 	}
 
-	dataTarGz, md5sums, instSize, err := createDataTarGz(info)
+	dataTarGz, md5sums, instSize, dataTarballName, err := createDataTarball(info)
```
Comment (caarlos0) [inline deb/deb.go:94]: `dataTarGz` could be named `dataTarball` now I think

**Review by caarlos0 — APPROVED** [review]
minor nitpick, lgtm otherwise

great work, thanks!

File: deb/deb.go (hunk 1 above)
Comment (erikgeiser) [reply deb/deb.go]: Thanks, I found a lot of other places too where I did not update the naming. It should all be consistent now.

**Review by caarlos0 — APPROVED** [review]

**Comment by erikgeiser** [comment]
I've prepared another PR that changes the default compression algorithm and the corresponding documentation. I will publish it as soon as this PR is merged, so you can easily update the default in another release if everything goes well.

**Pull request merged**